### PR TITLE
ProcessButtonPresses and fixes

### PR DESCRIPTION
**Pull request by erorcun**
Converted some array references to arrays to allow ASan to do it's job.

### CODE REVIEW

File: src/core/Frontend.cpp [hunk 1]
```diff
@@ -65,7 +68,7 @@ float &CMenuManager::menuXYpadding = *(float*)0x5F355C;	// don't know the origin
 float &CMenuManager::actionTextScaleX = *(float*)0x5F2E40;
 float &CMenuManager::actionTextScaleY = *(float*)0x5F2E44;
 
-int& CMenuManager::sthWithButtons = *(int32*)0x8E2880;
+int &CMenuManager::sthWithButtons = *(int32*)0x8E2880;
```
Comment (ShFil119) [inline src/core/Frontend.cpp]: first type should be also `int32`

File: src/core/Frontend.h [hunk 2]
```diff
@@ -399,35 +409,35 @@ class CMenuManager
 	bool m_bShowMouse;
 	tSkinInfo m_sSkin;
 	tSkinInfo *m_pSelectedSkin;
- tSkinInfo *field_438;
- float field_43C;
-	int m_nCurrExSize;
+	int32 m_nFirstVisibleRowOnList;
+	float m_nCurListItemY;
+	int m_nTotalListRow;
```
Comment (ShFil119) [inline src/core/Frontend.h]: `int32`

File: src/core/Frontend.h [hunk 3]
```diff
@@ -399,35 +409,35 @@ class CMenuManager
...
 	int m_nSkinsTotal;
  char _unk0[4];
-	int m_nCurrExOption;
+	int m_nSelectedListRow;
```
Comment (ShFil119) [inline src/core/Frontend.h]: also `int32`

File: src/core/Frontend.cpp [hunk 4]
```diff
@@ -1834,255 +1838,1207 @@ void CMenuManager::Process(void)
...
+				CMenuManager::m_PrefsSFXVolume = CMenuManager::m_PrefsSFXVolume - 8;
+				if (CMenuManager::m_PrefsSFXVolume < 0) {
+					CMenuManager::m_PrefsSFXVolume = 0;
+				}
+				if (0x7e < CMenuManager::m_PrefsSFXVolume) {
+					CMenuManager::m_PrefsSFXVolume = 0x7f;
+				}
+				DMAudio.SetEffectsMasterVolume(uchar)(CMenuManager::m_PrefsSFXVolume);
+				SaveSettings();
+				break;
```
Comment (ShFil119) [inline src/core/Frontend.cpp]: imho decimal would be better

Comment (erorcun) [reply src/core/Frontend.cpp]: i'm waiting for others to comment on whether I should remove that part (on discord)

File: src/core/Frontend.h (hunk 3 above)
Comment (erorcun) [reply src/core/Frontend.h]: ok

File: src/core/Frontend.h (hunk 2 above)
Comment (erorcun) [reply src/core/Frontend.h]: ok

File: src/core/Frontend.cpp (hunk 4 above)
Comment (erorcun) [reply src/core/Frontend.cpp]: done

**Pull request merged**